sputnikdao2/tests/test_quit.rs

```diff
@@ -0,0 +1,246 @@
+    assert_eq!(change_policy, 1);
+    call!(
+        root,
+        dao.act_proposal(change_policy - 1, Action::VoteApprove, None)
+    )
+    .assert_success();
+
+    let add_to_roles = |user: &UserAccount, roles: Vec<&str>| {
+        for role in roles {
+            add_member_to_role_proposal(&root, &dao, user.account_id.clone(), role.to_string())
+                .assert_success();
+
+            // approval
+            let proposal = view!(dao.get_last_proposal_id()).unwrap_json::<u64>();
+            call!(
+                root,
+                dao.act_proposal(proposal - 1, Action::VoteApprove, None)
+            )
+            .assert_success();
+        }
+    };
+    add_to_roles(&user2, vec!["has_2", "has_23", "has_234"]);
+    add_to_roles(&user3, vec!["has_3", "has_23", "has_234"]);
+    add_to_roles(&user4, vec!["has_234"]);
+
+    let role_members = |role_permission: &sputnikdao2::RolePermission| -> Vec<AccountId> {
+        if let RoleKind::Group(ref members) = role_permission.kind {
+            let mut members = members.into_iter().cloned().collect::<Vec<_>>();
+            members.sort();
+            members
+        } else {
+            vec![]
+        }
+    };
+
+    // quits and returns the remaining role names and their members
+    // this is a Vec so the order is preserved
+    type RoleNamesAndMembers = Vec<(String, Vec<AccountId>)>;
+    type RoleNamesAndMembersRef<'a> = Vec<(&'a str, Vec<&'a AccountId>)>;
+
+    // return role names and members form a dao
+    let dao_roles = || -> RoleNamesAndMembers {
+        view!(dao.get_policy())
+            .unwrap_json::<Policy>()
+            .roles
+            .into_iter()
+            .map(|role_permission| (role_permission.name.clone(), role_members(&role_permission)))
+            .collect()
+    };
+
+    // makes references into a RoleNamesAndMembers
+    // so they are easier to compare against
+    fn dao_roles_ref(dao_roles: &RoleNamesAndMembers) -> RoleNamesAndMembersRef {
+        dao_roles
+            .iter()
+            .map(|(name, members)| (name.as_str(), members.iter().collect()))
+            .collect::<Vec<(&str, Vec<&AccountId>)>>()
+    }
+
+    let quit =
+        |user: &UserAccount, user_check: &UserAccount, dao_name: String| -> Result<bool, String> {
+            use near_sdk_sim::transaction::ExecutionStatus;
+            use near_sdk_sim::ExecutionResult;
+            let res: ExecutionResult = call!(
+                user,
+                dao.quit_from_all_roles(user_check.account_id.clone(), dao_name),
+                deposit = to_yocto("0")
+            );
+            match res.status() {
+                ExecutionStatus::SuccessValue(_bytes) => Ok(res.unwrap_json::<bool>()),
+                ExecutionStatus::Failure(err) => Err(err.to_string()),
+                _ => panic!("unexpected status"),
+            }
+        };
+
+    // initial check,
+    // when nobody has quit yet
+    let roles = dao_roles();
+    {
```

Comment:
Curious why this is in its own block here?

Reply:
I'll remove it, thx!
It's a leftover from before because there were more operations but I ended-up using the `dao_roles_ref`